stdlib/test/collections/test_list.mojo

```diff
@@ -286,6 +286,60 @@ def test_list_reverse_move_count():
     # Keep vec alive until after we've done the last `vec.data + N` read.
     _ = vec^
 
+def test_list_index():
+    var test_list_a = List[Int](10,20,30,40,50)
+    
+    # Basic Functionality Tests
+    assert_equal(test_list_a.index(10), 0)  # Verify finding the first element's index.
+    assert_equal(test_list_a.index(30), 2)  # Verify finding the index of a middle element.
+    assert_equal(test_list_a.index(50), 4)  # Verify finding the last element's index.
+    with assert_raises():  # Verify error for a non-existent element.
```

Comment:
**Suggestion** You could specify `contains=<some_msg>` as a keyword argument to the `assert_raises` to verify the error message if you'd like.  This would be nice if you adopt my earlier comment about including the `normalized_start` and length of the list in the error message.

Reply:
Done